### ISSUE TITLE
tests failed on armv7

### ISSUE THREAD

**Issue by juhp** [jgm/cmark-hs#6]
Not sure what is at fault but I note that some test-cases are failing when test building 0.5.1 in Fedora 24 development (Rawhide) for armv7hl:

```
:
+ ./Setup test
Running 1 test suites...
Test suite test-cmark: RUNNING...

Cases: 6  Tried: 0  Errors: 0  Failures: 0
Cases: 6  Tried: 1  Errors: 0  Failures: 0
Cases: 6  Tried: 2  Errors: 0  Failures: 0
Cases: 6  Tried: 3  Errors: 0  Failures: 0
Cases: 6  Tried: 4  Errors: 0  Failures: 0

### Failure in: 4
expected: Node (Just (PosInfo {startLine = 1, startColumn = 1, endLine = 1, endColumn = 13})) DOCUMENT [Node (Just (PosInfo {startLine = 1, startColumn = 1, endLine = 1, endColumn = 13})) PARAGRAPH [Node Nothing DOCUMENT [],Node Nothing EMPH [Node Nothing DOCUMENT []]]]
 but got: Node (Just (PosInfo {startLine = 1, startColumn = 1, endLine = 1, endColumn = 13})) DOCUMENT [Node (Just (PosInfo {startLine = 1, startColumn = 1, endLine = 1, endColumn = 13})) PARAGRAPH [Node Nothing (TEXT "Hello ") [],Node Nothing EMPH [Node Nothing (TEXT "world") []]]]

Cases: 6  Tried: 5  Errors: 0  Failures: 1

### Failure in: 5
expected: "> Hello\n> *world*\n"
 but got: "> **\n"
Cases: 6  Tried: 6  Errors: 0  Failures: 2
Test suite test-cmark: FAIL
:
```

This is with ghc-7.8.4.

**Comment by jgm** [comment]
Very strange.  Have you tried building in the same
environment on non-armv7 machines?

+++ Jens Petersen [Feb 04 16 22:08 ]:

**Comment by juhp** [comment]
Yes: http://koji.fedoraproject.org/koji/taskinfo?taskID=12881418

**Comment by jgm** [comment]
Can you try building jgm/cmark (the C library cmark-hs
wraps) on your armv7 platform?  You'd just need cmake,
and python3 to run the tests.  It should be a
straightforward clone, make, make test.

+++ Jens Petersen [Feb 05 16 01:03 ]:

**Comment by juhp** [comment]
Yes, it builds fine and `make test` runs okay.

(For the time being I have disabled the cmark-hs tests on armv7.)

**Comment by jgm** [comment]
OK, so that suggests to me that the cmake build procedure is
doing something that needs to be done on armv7, and that
isn't being done by the simple build ghc performs.

+++ Jens Petersen [Feb 07 16 18:25 ]:

**Comment by juhp** [comment]
Edit: just to be clear - the cmark-hs tests still fail.

Note I am actually building the Haskell binding with my pkgconfig patch (thanks for the merge).
But this problem occurs in either case anyway.  I suspect it could be either some issue with armv7 ghc or some endianness etc with the Haskell binding possibly?  I haven't looked at the tests carefully so I am not sure what the failures are about exactly.

**Comment by juhp** [comment]
I can't reproduce anymore with 0.5.6 and bundled cmark in Fedora 28 Rawhide FWIW.
Not sure what changed, but the ghc version is now 8.0.2.

Neither with 0.5.5 and system cmark.
